### ISSUE TITLE
Help migrate 🍉 docs from `mdbook` to Docusaurus

### ISSUE THREAD

**Issue by radex** [Nozbe/WatermelonDB#1481]
Hey,

I'd like to migrate WatermelonDB docs from `mdbook` to Docusaurus, a de facto standard in the React/React Native community for project websites. With that, I'd also like documentation to feature an actual API reference, and not just guides. I hope that will help 🍉 users a lot.

But with many other maintenance responsibilities on my mind, I'd like help from the community with this. Can you help?

**Comment by KrisLau** [comment]
I can help but would probably just need some direction since I've never done this before!

**Comment by ErickLuizA** [comment]
I want to help too, do you have anything in mind related to how it should be structured in the repo?

The current setup has the docs-master/ with the .md files and the generated docs/ folder.

Can it be changed to something like the structure found in the [react-native-navigation](https://github.com/wix/react-native-navigation/tree/master/website) repo, it features the API reference you mentioned or even [react-native-website](https://github.com/facebook/react-native-website)

Both of these examples also have versioning, is that something useful here?

**Comment by radex** [comment]
@ErickLuizA Yes, I'd like to change the structure a bit, it seems to confuse a lot of people that docs-master/ is the current version of docs that matches code on master, while `docs/` matches the latest published version.

Regarding versioning, it's a nice to have, but not super important to me. I think we can start with a simple docusaurus setup, then add API reference, and then maybe think about how to ship versioned docs.

**Comment by radex** [comment]
We have this! @ErickLuizA thank you for your contribution.

The new website will be shipped alongside 0.26.0 release, for now you can view it locally like so:

```
cd docs-website
yarn
cd ..
yarn docs:dev  # opens in browser
```

**Comment by radex** [comment]
@KrisLau Your help would be definitely appreciated.

One thing you could start with if you're unsure about drafting new articles is just cleaning up the existing docs:

- Splitting large guide articles into smaller ones
- Throughout the docs, adding types to examples would be helpful. Things have changed since 2018, and I think we can now assume that most people use TypeScript/Flow, and whoever doesn't can just ignore them
- Changing navigation to be closer to your suggestion seems like a good idea

**Comment by KrisLau** [comment]
@radex Will do!

**Comment by radex** [comment]
> About the generate docs from source code, there are some solutions, I tested the [jsdoc-to-mdx](https://github.com/naver/jsdoc-to-mdx), it appears to work, although I kind of just put some text and see if it outputs something, if you wanna check it out, it's in this [branch](https://github.com/ErickLuizA/WatermelonDB/tree/migrate-docs-docusaurus-jsdoc), btw, for some reason, to work I need to put the comment in the .js and d.ts to work 🤔

> There is [this gist](https://gist.github.com/slorber/0bf8c8c8001505f0f99a062ac55bf442) from the maintainer of docusaurus, but I was not able to make it run, there is also https://github.com/facebook/docusaurus/issues/3413, but it's for typescript.

@ErickLuizA Can you help out with that? In general, JSDoc or something similar to annotate symbols with text seems good. JSDoc has the advantage that is has been around for a very long time, so even if one breaks, there's always going to be some tool to extract it and format it somehow.

It would be great not to have to duplicate types and just take function signatures automatically, and I think I've seen somewhere one-off scripts to help with that for Flow, but Flow just isn't that popular anymore, so it might be hard to maintain. Typedoc on the other hand looks interesting (though I don't really like how it formats the output), but I'm not sure it's a good idea to keep the main in-code documentation not in the main code but in the d.ts files.

So I think it would be great to start with a working demo of jsdoc-to-mdx, and then iterate from there

**Comment by radex** [comment]
I added some JSDoc comments to Database, Collection, etc. cc @ErickLuizA

**Comment by radex** [comment]
Also to Model, Schema, sync

**Comment by ErickLuizA** [comment]
I have not been able to open a PR with the API Reference because I didn't find a way to make jsdoc-to-mdx work properly
I don't know why, but it has some problems with flow or the .d.ts files or both.

I will try to work on another solution, maybe with [jsdoc-to-markdown](https://github.com/jsdoc2md/jsdoc-to-markdown), it does not work out of the box with docusaurus but it's more popular, I will see if it can work.

**Comment by radex** [comment]
@lucaswitch Much appreciated, do you need guidance as for where to start?

**Comment by radex** [comment]
I added JSDoc to Query class as well

**Comment by radex** [comment]
@KrisLau Nice, appreciated. A few nitpicks and the README, and we'll get that merged.

**Comment by adblanc** [comment]
Hi, would love to help if still possible!

I don't know if that's the appropriate place to say this but I just tried the lib for the first time and I must say that getting a new app to work correctly with 🍉 following the documentation was a pain.
Maybe that's just me but I really struggled to find informations about `DatabaseProvider`, `withDatabase` and even the examples in the repo didn't help.

Anyway, let me know if I can contribute!

**Comment by radex** [comment]
@adblanc You can!

**Comment by lucaswitch** [comment]
@radex I was busy, is there any new docs job left to do?

**Comment by lucaswitch** [comment]
I'm starting on the task "Add TypeScript types to examples in the docs" not quite done yet.

**Comment by Wanderson77** [comment]
@radex 
Hello everyone, I'm studying React Native and Watermelon and would also like to help with project documentation. Do you have any tips on where to start?

**Comment by Ich-bin-naoya** [comment]
I think It might be a good idea to write a way to test with jest and WatermelonDB. I put a memo under [my blog](https://medium.com/@naoyam_m/a-way-to-test-with-watermelondb-28ae275957bc). If it is ok to add, I will do that!

**Comment by maxwhipw** [comment]
Suggestion to make a WIP version of the docs available outside of the repo or to prioritize launching what you have.

Reason being is the JS ecosystem is heading towards Vite and has been at Typescript and from what I can tell Watermelon can work with both of these but it's not apparent from the website that is shown in the repo and pops up on search engines.

I think this will greatly hurt adoption and give people the wrong impression, more than having incomplete docs.

In my opinion this project has the potential to be very important in the ecosystem as sqlite is just perfect to pair with Electron like apps, and people are rediscovering why local first is so valuable.

I hope to be working with watermelon for a while and I'm fairly good at writing docs and processes so if opportunity comes up to contribute I might be able to help.
